darwin.adv_cmds fails to build

**Describe the bug**

The [`darwin.adv_cmds`](https://github.com/NixOS/nixpkgs/blob/47f61c9d7f116701ad9ca0d4b894ebb5f36ef5cf/pkgs/os-specific/darwin/apple-source-releases/default.nix#L243) package fails to build on both `master`, `release-20.03` and `release-19.09`.

Hydra:

* https://hydra.nixos.org/job/nixpkgs/nixpkgs-20.03-darwin/darwin.adv_cmds.x86_64-darwin

* https://hydra.nixos.org/job/nixpkgs/trunk/darwin.adv_cmds.x86_64-darwin

**To Reproduce**
```
$ cd nixpkgs
$ git checkout master
$ git pull
$ nix-build -A darwin.adv_cmds
...
    /nix/store/y4k1kzwlpxi5gwi9d8854cika2crypmq-Toolchains/XcodeDefault.xctoolchain/usr/bin/dsymutil /private/var/folders/5h/b11m6fxj2tqgbxwz5bgjy42c0000gn/T/nix-build-adv_cmds-osx-10.11.6.drv-0/adv_cmds-163/Products/Release/locale -o /private/var/folders/5h/b11m6fxj2tqgbxwz5bgjy42c0000gn/T/nix-build-adv_cmds-osx-10.11.6.drv-0/adv_cmds-163/Products/Release/locale.dSYM
error: cannot parse the debug map for '/private/var/folders/5h/b11m6fxj2tqgbxwz5bgjy42c0000gn/T/nix-build-adv_cmds-osx-10.11.6.drv-0/adv_cmds-163/Products/Release/locale': No such file or directory

** BUILD FAILED **

The following build commands failed:
    GenerateDSYMFile /private/var/folders/5h/b11m6fxj2tqgbxwz5bgjy42c0000gn/T/nix-build-adv_cmds-osx-10.11.6.drv-0/adv_cmds-163/Products/Release/locale.dSYM /private/var/folders/5h/b11m6fxj2tqgbxwz5bgjy42c0000gn/T/nix-build-adv_cmds-osx-10.11.6.drv-0/adv_cmds-163/Products/Release/locale
(1 failure)
builder for '/nix/store/zipwrp6qvawym1y4wyznyqh55glwa0vb-adv_cmds-osx-10.11.6.drv' failed with exit code 1
error: build of '/nix/store/zipwrp6qvawym1y4wyznyqh55glwa0vb-adv_cmds-osx-10.11.6.drv' failed
```


**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.3`
 - channels(bas): `"nixpkgs-19.09pre191265.c4adeddb5f8"`
 - nixpkgs: `/Users/bas/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: darwin.adv_cmds
# a list of nixos modules affected by the problem
module:
```

CC @matthewbauer

--- a/pkgs/os-specific/darwin/apple-source-releases/adv_cmds/default.nix
+++ b/pkgs/os-specific/darwin/apple-source-releases/adv_cmds/default.nix
@@ -3,27 +3,22 @@
 appleDerivation {
   # We can't just run the root build, because https://github.com/facebook/xcbuild/issues/264
 
+  patchPhase = ''
+    substituteInPlace adv_cmds.xcodeproj/project.pbxproj \
+      --replace '/usr/lib/libtermcap.dylib' 'libncurses.dylib'
+  '';
+
   # pkill requires special private headers that are unavailable in
   # NixPkgs. These ones are needed:
   #  - xpc/xpxc.h
   #  - os/base_private.h
   #  - _simple.h
   # We disable it here for now. TODO: build pkill inside adv_cmds
-
-  # We also disable locale here because of some issues with a missing
-  # "lstdc++".
-  patchPhase = ''
-    substituteInPlace adv_cmds.xcodeproj/project.pbxproj \
-      --replace "FD201DC214369B4200906237 /* pkill.c in Sources */," "" \
-      --replace "FDF278D60FC6204E00D7A3C6 /* locale.cc in Sources */," "" \
-      --replace '/usr/lib/libtermcap.dylib' 'libncurses.dylib'
-  '';
-
   buildPhase = ''
     targets=$(xcodebuild -list \
                 | awk '/Targets:/{p=1;print;next} p&&/^\s*$/{p=0};p' \
                 | tail -n +2 | sed 's/^[ \t]*//' \
-                | grep -v -e Desktop -e Embedded -e mklocale -e colldef)
+                | grep -v -e Desktop -e Embedded -e mklocale -e pkill -e pgrep -e colldef)
 
     for i in $targets; do
       xcodebuild SYMROOT=$PWD/Products OBJROOT=$PWD/Intermediates -target $i
@@ -34,7 +29,7 @@ appleDerivation {
   installPhase = ''
     for f in Products/Release/*; do
       if [ -f $f ]; then
-        install -D $file $out/bin/$(basename $f)
+        install -D $f $out/bin/$(basename $f)
       fi
     done
 

